src/main/java/org/truffleruby/interop/ToJavaStringNode.java

```diff
@@ -40,8 +46,24 @@ String stringCached(DynamicObject value,
     }
 
     @Specialization(guards = "isRubyString(value)", replaces = "stringCached")
-    public String stringUncached(DynamicObject value) {
-        return StringOperations.getString(value);
+    public String stringUncached(DynamicObject value,
+            @Cached("createBinaryProfile()") ConditionProfile asciiOnlyProfile,
+            @Cached("create()") RopeNodes.BytesNode bytesNode) {
+        final Rope rope = StringOperations.rope(value);
+        final byte[] bytes = bytesNode.execute(rope);
+
+        if (asciiOnlyProfile.profile(rope.isAsciiOnly())) {
+            return RopeOperations.decodeAscii(bytes, 0, bytes.length);
+        } else {
+            return decodeNonAscii(rope, bytes);
```

Comment:
Why not call `RopeOperations.decodeNonAscii()` and make that public?
Or move the helper below in `RopeOperations`?
Getting the bytes again (`getBytes()`) behind a boundary should be pretty cheap compared to the decoding.

Reply:
I already had the bytes in hand. It seemed unnecessary to fetch them again.